### PR TITLE
Support for mysql 5.6

### PR DESCRIPTION
**Pull request by zageyiff**
I have modified the mysql-snmp in order to support mysql 5.6
I have forked and commit the changes, hope it helps!!

Regards

### CODE REVIEW

**Comment by masterzen** [comment]
Thank you very much for your contribution!

I've a couple of comments (inlined in the patch).

File: mysql-snmp [hunk 1]
```diff
@@ -129,11 +130,21 @@ sub parse_innodb_status {
...
+		}
+        elsif ($line =~ m/RW-shared spins/) {
+            push(@spin_waits,  $self->tonum($row[2]));
+            push(@spin_rounds, $self->tonum($row[4]));
+            push(@os_waits,    $self->tonum($row[7]));
+        }
+		elsif ($line =~ m/RW-excl spins/) {
+            push(@spin_waits,  $self->tonum($row[2]));
+            push(@spin_rounds, $self->tonum($row[4]));
+            push(@os_waits,    $self->tonum($row[7]));
```
Comment (masterzen) [inline mysql-snmp:156]: watch out for whitespace/tabs damages. The github diff output looks like the new code doesn't use the same standard as the old code.

File: mysql-snmp [hunk 2]
```diff
@@ -197,7 +208,7 @@ sub parse_innodb_status {
         }
         elsif ($line =~ m/Pending normal aio/) {
             $status{'pending_normal_aio_reads'}  = $self->tonum($row[4]);
-            $status{'pending_normal_aio_writes'} = $self->tonum($row[7]);
+            $status{'pending_normal_aio_writes'} = $self->tonum($row[12]);
```
Comment (masterzen) [inline mysql-snmp:226]: Was there a bug there in the original code?
This doesn't look like it is mysql 5.6 specific.

If yes, can you extract this in a different patch (you can keep it in the same pull request)?

File: mysql-snmp [hunk 3]
```diff
@@ -242,7 +253,7 @@ sub parse_innodb_status {
         elsif ($merged_op_seen && $line =~ m/ insert \d+, delete mark/) {
            #merged operations:
            # insert 0, delete mark 0, delete 0
-           $status{'ibuf_inserts'} = $self->tonum($1);
+           $status{'ibuf_inserts'} = $self->tonum($row[2]);
```
Comment (masterzen) [inline mysql-snmp:279]: Same as the comment on Pending normal aio

File: mysql-snmp [hunk 4]
```diff
@@ -276,6 +287,9 @@ sub parse_innodb_status {
            $status{'last_checkpoint'} = defined($row[4]) ? $self->make_bigint($row[3], $row[4]) : $self->tonum($row[3]);
         }
         # BUFFER POOL AND MEMORY
+		elsif ($line =~ m/^INDIVIDUAL BUFFER POOL INFO/) {
+			$individual_buffer_pool_info_seen = 1;
+		}
```
Comment (masterzen) [inline mysql-snmp]: whitespace/tab issue

File: tests/test.pl [hunk 5]
```diff
@@ -349,8 +349,8 @@ sub readfile
    $innodb_parser->parse_innodb_status(readfile('tests/data/issue5.txt')),
    {
        'spin_waits'                => '41',
-       'spin_rounds'               => '757',
-       'os_waits'                  => '1',
+       'spin_rounds'               => '1087',
+       'os_waits'                  => '9',
```
Comment (masterzen) [inline tests/test.pl:353]: so this test was failing for you, right?
I'm surprised I didn't catch those :)

**Comment by masterzen** [comment]
So if you could split the patch in two and extract the bits not related to mysql 5.6, and fix the indentations, I'd be more than happy to merge your contributions.

Thank you!

File: mysql-snmp (hunk 2 above)
Comment (zageyiff) [inline mysql-snmp:226]: The format for this line changed, so I'll fix it to make it backwards compatible and put in the code the line in a comment so its clear
Before:
 Pending normal aio reads: 0, aio writes: 0,
Newer:
 Pending normal aio reads: 0 [0, 0, 0, 0] , aio writes: 0 [0, 0, 0, 0] ,

File: tests/test.pl (hunk 5 above)
Comment (zageyiff) [inline tests/test.pl:353]: This was testing against incorrect values (but test passed), before there were four values in the same line:
m/RW-shared spins \d+, OS waits \d+; RW-excl/
In newer mysql these are different lines
RW-shared spins
RW-excl spins
So we were missing to add a value
And this issue5.txt had the format  like newer mysql, split in two lines

**Comment by zageyiff** [comment]
I'll fix the indentations and do a new commit. Everything is related to mysql 5.6 so it will be just one commit

**Comment by masterzen** [comment]
OK, thank you for your explanations.
I'll try to merge the revised patch this week.

**Comment by masterzen** [comment]
hmm, did you deliberately closed this PR?
I'm going to reopen so that I don't forgot to merge it later in the week :)

**Comment by zageyiff** [comment]
Yes, I closed it. I messed with the commits and lost the history. So I did a new pull request here
https://github.com/masterzen/mysql-snmp/pull/24

I'll be closing this one again :)
